WooCommerce/Classes/Authentication/WebAuth/WebKitViewController.swift

```diff
@@ -0,0 +1,519 @@
+        }
+
+        DispatchQueue.main.async {
+            let cookieStore = webView.configuration.websiteDataStore.httpCookieStore
+            authenticator.request(url: url, cookieJar: cookieStore) { (request) in
+                completion(request)
+            }
+        }
+    }
+}
+
+class WebKitViewController: UIViewController, WebKitAuthenticatable {
+    @objc let webView: WKWebView
+    @objc let progressView = WebProgressView()
+    @objc let titleView = NavigationTitleView()
+
+    @objc lazy var backButton: UIBarButtonItem = {
+        let button = UIBarButtonItem(image: UIImage.gridicon(.chevronLeft).imageFlippedForRightToLeftLayoutDirection(),
+                               style: .plain,
+                               target: self,
+                               action: #selector(goBack))
+        button.title = NSLocalizedString("Back", comment: "Previous web page")
+        return button
+    }()
+    @objc lazy var forwardButton: UIBarButtonItem = {
+        let button = UIBarButtonItem(image: .gridicon(.chevronRight),
+                               style: .plain,
+                               target: self,
+                               action: #selector(goForward))
+        button.title = NSLocalizedString("Forward", comment: "Next web page")
+        return button
+    }()
+    @objc lazy var shareButton: UIBarButtonItem = {
+        let button = UIBarButtonItem(image: .gridicon(.shareiOS),
+                               style: .plain,
+                               target: self,
+                               action: #selector(share))
+        button.title = NSLocalizedString("Share", comment: "Button label to share a web page")
+        return button
+    }()
+    @objc lazy var safariButton: UIBarButtonItem = {
+        let button = UIBarButtonItem(image: .gridicon(.globe),
+                               style: .plain,
+                               target: self,
+                               action: #selector(openInSafari))
+        button.title = NSLocalizedString("Safari", comment: "Button label to open web page in Safari")
+        button.accessibilityHint = NSLocalizedString("Opens the web page in Safari", comment: "Accessibility hint to open web page in Safari")
+        return button
+    }()
+    @objc lazy var refreshButton: UIBarButtonItem = {
+        let button = UIBarButtonItem(image: .gridicon(.refresh), style: .plain, target: self, action: #selector(WebKitViewController.refresh))
+        button.title = NSLocalizedString("Refresh", comment: "Button label to refresh a web page")
+        return button
+    }()
+    @objc lazy var closeButton: UIBarButtonItem = {
+        let button = UIBarButtonItem(image: .gridicon(.cross), style: .plain, target: self, action: #selector(WebKitViewController.close))
+        button.title = NSLocalizedString("webKit.button.dismiss", value: "Dismiss", comment: "Verb. Dismiss the web view screen.")
+        return button
+    }()
```

Comment:
While we don't need toolbar on the WebView yet (it's disabled by `secureInteraction` option), I decided to keep the (almost) full functionality. We can improve the preview feature by adding "device" button, already supported on WP-iOS with [PreviewWebKitViewController](https://github.com/wordpress-mobile/WordPress-iOS/blob/383304dc332f1b42db01b95c823b4a0392170907/WordPress/Classes/ViewRelated/Post/Preview/PreviewWebKitViewController.swift) subclass:

<img width=350 src="https://user-images.githubusercontent.com/3132438/199505079-1d65bcbb-7a9e-43be-92aa-cd375f0033b3.png">